packages/odyssey-storybook/src/components/odyssey-mui/DataTable/DataTable.stories.tsx

```diff
@@ -588,3 +601,299 @@ export const Empty: StoryObj<DataTableProps> = {
+      const filteredPlanetColumns = planetColumns.filter(
+        (column) => column.id !== "visit",
+      );
+
+      return [
+        ...filteredPlanetColumns,
+        {
+          id: "visit",
+          label: "Type of visit 🚀",
+          render: (updateFilters: UpdateFiltersCallback) => (
+            <Box>
+              <Button
+                variant="secondary"
+                label="🛬 Landing"
+                onClick={() =>
+                  updateFilters({ filterId: "visit", value: "landing" })
+                }
+              />
+              <Button
+                variant="secondary"
+                label="🛰️ Orbit"
+                onClick={() =>
+                  updateFilters({ filterId: "visit", value: "orbit" })
+                }
+              />
+              <Button
+                variant="secondary"
+                label="🛸 Flyby"
+                onClick={() =>
+                  updateFilters({ filterId: "visit", value: "flyby" })
+                }
+              />
+              <Button
+                variant="secondary"
+                label="🤷‍♂️ Any"
+                onClick={() => updateFilters({ filterId: "visit", value: "" })}
+              />
+            </Box>
+          ),
+        },
+      ];
+    }, [planetColumns]);
+
+    return (
+      <DataTable
+        {...props}
+        columns={planetColumns}
+        filters={filters}
+        getData={getData}
+        onReorderRows={onReorderRows}
+        onChangeRowSelection={onChangeRowSelection}
+      />
+    );
+  },
+};
+
+export const CustomFilterWithDefaultVariant: StoryObj<DataTableProps> = {
+  args: {
+    hasChangeableDensity: true,
+    hasColumnResizing: true,
+    hasColumnVisibility: false,
+    hasFilters: true,
+    hasPagination: false,
+    hasRowSelection: true,
+    hasSearch: true,
+    hasSorting: true,
+    hasRowReordering: false,
+  },
+  render: function C(props) {
+    const [data, setData] = useState<Planet[]>(planetData);
+
+    const getData = useCallback(
+      ({ ...props }: DataTableGetDataType) => {
+        return filterData({ data, ...props });
+      },
+      [data],
+    );
+
+    const onReorderRows = useCallback(
+      ({ ...props }: DataTableOnReorderRowsType) => {
+        const reorderedData = reorderData({ data, ...props });
+        setData(reorderedData as Planet[]);
+      },
+      [data],
+    );
+
+    const onChangeRowSelection = useCallback(
+      (rowSelection: DataTableRowSelectionState) => {
+        if (Object.keys(rowSelection).length > 0) {
+          console.log(`${Object.keys(rowSelection).length} selected`);
+        }
+      },
+      [],
+    );
+
+    const filters = useMemo(() => {
+      return [
+        ...planetColumns,
+        {
+          id: "startLetter",
+          label: "Starting letter",
+          variant:
+            "select" as DataTableColumn<DataTableRowData>["filterVariant"],
+          options: [
+            {
```

Comment:
Why the `as` on line `873`? Aren't variants known? We can type the object we're memoizing `useMemo<DataTableColumn<DataTableRowData>>` right?